2015/proposals/girdhar-yashasvi.md

```diff
@@ -0,0 +1,254 @@
+#Result-aggregation server for the installation-test scripts
+
+##Personal Details:
+
+* Name: Yashasvi Girdhar
+* Email: yash.girdhar@gmail.com
+* Personal Webpage : http://about.me/yashgirdhar
+* Github profile : https://github.com/itsyash
+* IRC nick (on freenode) : yash_
+* Telephone: +91-8143914414
+* Country of residence: India
+* Timezone: IST (GMT+5:30)
+* Primary language: English
+
+##Abstract:
+
+Software Carpentry has installation-test scripts which are used by students to check if they have correctly installed the required software for a workshop.
+
+The script basically performs a test for each software and outputs the result. In case of a failure, it prints the cause of failure and some diagnostic information on how to install the software correctly.
+Until now, we do not collect these results anywhere. But there are number of reasons why we should be doing that.
+
+Some sample queries that the system would serve will look like:
+
+* How many tests failed for a given workshop for students running Windows.
+* What fraction of failed tests were on OS X for a specific time period.
+* For a given workshop, find number of failed version tests for Git.
+
+Statistics like these will help us figure out more information about:
+* How much trouble the students are having with the installation.
+* Which dependencies need more of our attention (deprecating their older versions or adapting with the new versions).
+
+##Technical Details:
```

Comment:
I actually gave it a thought while writing my proposal but considering the independence between the backend of amy and this project (amy is for workshop details and this will store the test script results and manipulate them), I suppose we would anyways be making a separate module for this in Amy. If that is the case, I thought it would be good to make it an independent project. 
In future, we can always use something from Amy if we want in this project, such as fetching the workshop requirements from Amy.